### PR TITLE
Access Passes

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Adds Access Passes. They act like regular IDs, but when smacked onto another ID they will download the data onto the other ID.
How to use them:
* Put Access Pass into the Console
* Change access, job name or ID name
* Slap onto another ID. The ID will be updated.

Replaces access cards (deputy and mining, to be more specific)

## Why It's Good For The Game

Makes HoP's job easier and (somewhat) safer. 

## Changelog
:cl:
add: Added access passes! 
add: HoP now starts with a box of access passes in their backpack.
add: Maints access passes have a chance to spawn as maintenance loot
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
This is a good idea, but made completely redundant by the fact that heads get ID consoles in their offices to give access

**Comment by eeSPee** [comment]
Who has time to run at the console, log in, yadayada. Just give the sucker a access pass and be on your way!

**Comment by ItsTisme** [comment]
Problem is even with these passes unless I carry them with me I still have to walk them to my office which is the annoying part, only seems like a excuse for more traitors to break into my office

**Comment by bloons3** [comment]
What's the point of having an ID console, or even HoP anymore?
The closest analog to these would be the mining conscription kit, but that's specifically a Cargo/QM thing.

**Comment by eeSPee** [comment]
Ok. Removed the upgrade passes in head's lockers.

HoP still gets the boxes of passes cards tho. They have no access, they're just for making his job easier.

**Comment by FortSumter** [comment]
A better idea would be for the HOP to hand these out instead of ID updates as a separate item. It could have a number of uses and/or set to a certain department. This would give the players the extra access without permanently updating their IDs and without the HOP having to go track them down again to remove the access once they did what they needed. You can even use the sprite from NSV for the Squad lanyards and have them equip-able in the tie-slot.

As an example, there are no engineers and the SM needs setting up. An assistant knows how to do it and wants engineering access. You as the HOP would rather not give the assistant permanent engineering access to just do the one thing, and so you hand them one of these lanyards for engineering. Once they're done, they hand back the lanyard (or perhaps in a later PR you can give the HOP the ability to remotely invalidate the lanyard access so the assistant doesn't even have to return it).

TLDR make it a hall-pass instead of an ID update.

**Comment by eeSPee** [comment]
> A better idea would be for the HOP to hand these out instead of ID updates as a separate item. It could have a number of uses and/or set to a certain department. This would give the players the extra access without permanently updating their IDs and without the HOP having to go track them down again to remove the access once they did what they needed. You can even use the sprite from NSV for the Squad lanyards and have them equip-able in the tie-slot.
> 
> As an example, there are no engineers and the SM needs setting up. An assistant knows how to do it and wants engineering access. You as the HOP would rather not give the assistant permanent engineering access to just do the one thing, and so you hand them one of these lanyards for engineering. Once they're done, they hand back the lanyard (or perhaps in a later PR you can give the HOP the ability to remotely invalidate the lanyard access so the assistant doesn't even have to return it).
> 
> TLDR make it a hall-pass instead of an ID update.

What makes you think they will return them, though?

**Comment by FortSumter** [comment]
On paper, they would return them because they were given the pass for a reason and should return it after they are done for RP. In reality, you and I both know they won't, but it would be easier and less of a pain in the ass to simply stun the degenerate and remove the lanyard from their necks than take their IDs, go back to your office to remove their permanent access, and then handing it back to them. This also sets up for another PR in which the HOP gets the ability to remotely manage the lanyards either via a console or his PDA like he can in NSV for squads. If you want to go that extra mile in this PR you can, I just thought having the concept in first to test it before expanding on the idea especially my idea is a significant departure from what you had in mind originally.

File: code/game/objects/items/cards_ids.dm [hunk 1]
```diff
@@ -757,3 +757,35 @@ update_label("John Doe", "Clowny")
...
+			idcard.name = name
+		to_chat(user, "You upgrade your [idcard] with the [name].")
+		log_id("[key_name(user)] added access to '[idcard]' using [src] at [AREACOORD(user)].")
+		qdel(src)
+
+/obj/item/card/id/pass/minisec
+	name = "deputy promotion card"
+	desc = "Swipe it on your ID card to become a Deputy. How does it feel to be at the other side of the stunbaton?"
+	assignment = "Deputy"
+	access = list (ACCESS_SEC_DOORS,ACCESS_MAINT_TUNNELS,ACCESS_COURT,ACCESS_BRIG,ACCESS_WEAPONS)
```
Comment (utilbird) [inline code/game/objects/items/cards_ids.dm:786]: Don't need to separate if/for/list and their arguments with a space

Comment (eeSPee) [reply code/game/objects/items/cards_ids.dm:786]: Sorry. Habbit from C#.

**Comment by PowerfulBacon** [comment]
stale
